backend/btrixcloud/models.py

```diff
@@ -1799,3 +1977,220 @@ class PageOutWithSingleQA(Page):
+
+
+# ============================================================================
+class SuccessResponse(BaseModel):
+    """Response for API endpoints that return success"""
+
+    success: bool
+
+
+# ============================================================================
+class SuccessResponseStorageQuota(SuccessResponse):
+    """Response for API endpoints that return success and storageQuotaReached"""
+
+    storageQuotaReached: bool
+
+
+# ============================================================================
+class StartedResponse(BaseModel):
+    """Response for API endpoints that start crawls"""
+
+    started: str
+
+
+# ============================================================================
+class AddedResponse(BaseModel):
+    """Response for API endpoints that return added"""
+
+    added: bool
+
+
+# ============================================================================
+class AddedResponseId(AddedResponse):
+    """Response for API endpoints that return added + id"""
+
+    id: UUID
+
+
+# ============================================================================
+class AddedResponseName(AddedResponse):
+    """Response for API endpoints that add resources and return name"""
+
+    name: str
+
+
+# ============================================================================
+class AddedResponseIdQuota(AddedResponse):
+    """Response for API endpoints that return str id and storageQuotaReached"""
+
+    id: str
+    storageQuotaReached: bool
+
+
+# ============================================================================
+class AddedResponseIdName(AddedResponse):
+    """Response for API endpoints that add resource and return id and name"""
+
+    id: UUID
+    name: str
+
+
+# ============================================================================
+class DeletedResponse(BaseModel):
+    """Response for delete API endpoints"""
+
+    deleted: bool
+
+
+# ============================================================================
+class DeletedResponseQuota(DeletedResponse):
+    """Response for delete API endpoints"""
+
+    storageQuotaReached: bool
+
+
+# ============================================================================
+class DeletedCountResponse(BaseModel):
+    """Response for delete API endpoints that return count"""
+
+    deleted: int
+
+
+# ============================================================================
+class RemovedResponse(BaseModel):
+    """Response for API endpoints for removing resources"""
+
+    removed: bool
+
+
+# ============================================================================
+class EmptyResponse(BaseModel):
+    """Response for API endpoints that return nothing"""
+
+
+# ============================================================================
+
+### SPECIFIC PAGINATED RESPONSE MODELS ###
+
+
+# ============================================================================
+class PaginatedBackgroundJobResponse(PaginatedResponse):
```

Comment:
This would be a good use case for generics, so don't hace to duplicate all of these types?
Though could be done as a follow-up probably